src/main/java/mybatis/com/mybatis/Dto/SubjectCreationDto.java

```diff
@@ -0,0 +1,11 @@
+package mybatis.com.mybatis.Dto;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import lombok.Data;
+import mybatis.com.mybatis.Entity.TeacherEntity;
+
+@Data
+public class SubjectCreationDto {
+    private String name;
+    private TeacherCreationDto TeacherCreationDto;
```

Comment:
Do we need to accept the entire TeacherCreationDTO?
Aren't we create the teacher entity using the POST teacher API call

Reply:
no

Reply:
done, accepting teacherId instead of whole teacherCreationDto